R/zchunk_L201.socioeconomics_USA.R

```diff
@@ -0,0 +1,132 @@
+#' original data system was \code{L201.socioeconomics_USA.R} (gcam-usa level2).
+#' @details Interest rate, population, and GDP for GCAM-USA.
+#' @importFrom assertthat assert_that
+#' @importFrom dplyr filter mutate select
+#' @importFrom tidyr gather spread
+#' @author RLH October 2017
+module_gcam.usa_L201.socioeconomics_USA <- function(command, ...) {
+  if(command == driver.DECLARE_INPUTS) {
+    return(c(FILE = "gcam-usa/states_subregions",
+             "L100.Pop_thous_state",
+             "L100.GDP_mil90usd_state",
+             "L102.pcgdp_thous90USD_GCAM3_ctry_Y"))
+  } else if(command == driver.DECLARE_OUTPUTS) {
+    return(c("L201.InterestRate_GCAMUSA",
+             "L201.Pop_GCAMUSA",
+             "L201.BaseGDP_GCAMUSA",
+             "L201.LaborForceFillout_GCAMUSA",
+             "L201.LaborProductivity_GCAMUSA"))
+  } else if(command == driver.MAKE) {
+
+    # silence package checks
+    year <- value <- state <- totalPop <- baseGDP <- iso <- growth <- timestep <- NULL
+
+    all_data <- list(...)[[1]]
+
+    # Load required inputs
+    states_subregions <- get_data(all_data, "gcam-usa/states_subregions")
+    L100.Pop_thous_state <- get_data(all_data, "L100.Pop_thous_state")
+    L100.GDP_mil90usd_state <- get_data(all_data, "L100.GDP_mil90usd_state")
+    L102.pcgdp_thous90USD_GCAM3_ctry_Y <- get_data(all_data, "L102.pcgdp_thous90USD_GCAM3_ctry_Y")
+
+    # ===================================================
+    # L201.InterestRate: Interest rates by region
+    L201.InterestRate <- tibble(region = states_subregions$state, interest.rate = socioeconomics.DEFAULT_INTEREST_RATE)
+
+    # L201.Pop_GCAMUSA: Population by region from the GCAM 3.0 core scenario
+    L201.Pop_GCAMUSA <- L100.Pop_thous_state %>%
+      filter(year %in% MODEL_YEARS) %>%
+      rename(totalPop = value,
+             region = state) %>%
+      mutate(totalPop = round(totalPop, socioeconomics.POP_DIGITS))
+
+    # L201.BaseGDP_GCAMUSA: Base GDP for GCAM-USA scenario
+    L201.BaseGDP_GCAMUSA <- L100.GDP_mil90usd_state %>%
+      filter(year == min(MODEL_YEARS)) %>%
+      rename(baseGDP = value,
+             region = state) %>%
+      mutate(baseGDP = round(baseGDP, socioeconomics.GDP_DIGITS)) %>%
+      select(-year)
+
+    # L201.LaborForceFillout: Labor force participation and productivity for all scenarios
+    # NOTE: No model of labor force used; labor force participation set to a constant
+    L201.LaborForceFillout <- tibble(region = states_subregions$state,
+                                     year.fillout = min(MODEL_YEARS),
+                                     laborforce = socioeconomics.DEFAULT_LABORFORCE)
+
+    # LABOR PRODUCTIVITY GROWTH RATE CALCULATION
+    # Labor productivity growth is calculated from the change in per-capita GDP ratio in each time period
+    # Calculate the growth rate in per-capita GDP
+    L201.LaborProductivity_GCAMUSA <- L102.pcgdp_thous90USD_GCAM3_ctry_Y %>%
+      filter(iso == "usa",
```

Comment:
Is this guaranteed to always be lower case? Maybe `tolower(iso)` is safer? Not sure.